src/uv_snapshot.c

```diff
@@ -677,17 +674,17 @@ int UvSnapshotPut(struct raft_io *io,
     cursor = &put->meta.header[1];
     bytePut64(&cursor, crc);
 
-    /* If the trailing parameter is set to 0, it means that we're restoring a
+    /* - If the trailing parameter is set to 0, it means that we're restoring a
      * snapshot. Submit a barrier request setting the next append index to the
-     * snapshot's last index + 1. */
-    if (trailing == 0) {
-        rv = UvBarrier(uv, snapshot->index + 1, &put->barrier,
-                       uvSnapshotPutBarrierCb);
-        if (rv != 0) {
-            goto err_after_configuration_encode;
-        }
-    } else {
-        uvSnapshotPutStart(put);
+     * snapshot's last index + 1.
+     * - When we are only writing a snapshot during normal operation, we close
+     * all current open segments. New writes can continue on newly opened
+     * segments that will only contain entries that are newer than the snapshot,
+     * and we don't change append_next_index. */
+    raft_index next_index = trailing == 0 ? snapshot->index + 1 : uv->append_next_index;
```

Comment:
Nitpick: I'd recommend to put all variable declarations at the top of the block, instead of inline. I've personally found that it helps with readability and also it reduces the risk of shadowing a variable previously declared in an outer block. That's the current convention in the code base IIRC.

Reply:
fixed